plugins/outputs/azure_data_explorer/azure_data_explorer.go

```diff
@@ -75,22 +65,38 @@ func (adx *AzureDataExplorer) Connect() error {
 		Authorizer: authorizer,
 	}
 	client, err := kusto.New(adx.Endpoint, authorization)
+	adx.Log.Debug("Client initialized successfully")
 
 	if err != nil {
 		return err
 	}
 	adx.client = client
-	adx.ingesters = make(map[string]localIngestor)
-	adx.createIngestor = createRealIngestor
-
+	adx.ingestors = make(map[string]ingest.Ingestor)
 	return nil
 }
 
 func (adx *AzureDataExplorer) Close() error {
+	var errs []error
+	for _, v := range adx.ingestors {
+		if err := v.Close(); err != nil {
+			// accumulate errors while closing ingestors
+			errs = append(errs, err)
+		}
+	}
+	if err := adx.client.Close(); err != nil {
+		errs = append(errs, err)
+	}
+
 	adx.client = nil
-	adx.ingesters = nil
+	adx.ingestors = nil
 
-	return nil
+	if len(errs) == 0 {
+		adx.Log.Info("Closed ingestors and client")
+		return nil
+	}
+
+	// Combine errors into a single object and return the combined error
+	return kustoerrors.GetCombinedError(errs...)
```

Comment:
Can we please invert this
```suggestion
	if len(errs) > 0 {
		// Combine errors into a single object and return the combined error
		return kustoerrors.GetCombinedError(errs...)
	}
	
	adx.Log.Info("Closed ingestors and client")
	return nil
```
This makes it easier to extend that function later and also eases following the flow.